src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -0,0 +1,377 @@
+    UUID holdingsWithPermLocation = createInstanceAndHolding(mainLibraryLocationId);
+    UUID holdingsWithTempLocation = createInstanceAndHolding(mainLibraryLocationId, annexLibraryLocationId);
+
+    Item itemWithHoldingPermLocation = buildItem(holdingsWithPermLocation, null, null);
+    Item itemWithHoldingTempLocation = buildItem(holdingsWithTempLocation, null, null);
+    Item itemWithTempLocation = buildItem(holdingsWithPermLocation, onlineLocationId, null);
+    Item itemWithPermLocation = buildItem(holdingsWithTempLocation, null, secondFloorLocationId);
+    Item itemWithAllLocation = buildItem(holdingsWithTempLocation, onlineLocationId, secondFloorLocationId);
+
+    Item[] itemsToCreate = {itemWithHoldingPermLocation, itemWithHoldingTempLocation,
+      itemWithTempLocation, itemWithPermLocation, itemWithAllLocation};
+
+    for (Item item : itemsToCreate) {
+      IndividualResource createdItem = createItem(item);
+      assertTrue(createdItem.getJson().containsKey("effectiveLocationId"));
+    }
+
+    Items mainLibraryItems = findItems("effectiveLocationId=" + mainLibraryLocationId);
+    Items annexLibraryItems = findItems("effectiveLocationId=" + annexLibraryLocationId);
+    Items onlineLibraryItems = findItems("effectiveLocationId=" + onlineLocationId);
+    Items secondFloorLibraryItems = findItems("effectiveLocationId=" + secondFloorLocationId);
+
+    assertEquals(1, mainLibraryItems.getTotalRecords().intValue());
+    assertThat(mainLibraryItems.getItems().get(0).getId(), is(itemWithHoldingPermLocation.getId()));
+
+    assertEquals(1, annexLibraryItems.getTotalRecords().intValue());
+    assertThat(annexLibraryItems.getItems().get(0).getId(), is(itemWithHoldingTempLocation.getId()));
+
+    assertEquals(2, onlineLibraryItems.getTotalRecords().intValue());
+
+    assertThat(onlineLibraryItems.getItems()
+        .stream()
+        .map(Item::getId)
+        .collect(Collectors.toList()),
+      hasItems(itemWithTempLocation.getId(), itemWithAllLocation.getId()));
+
+    assertEquals(1, secondFloorLibraryItems.getTotalRecords().intValue());
+    assertThat(secondFloorLibraryItems.getItems().get(0).getId(), is(itemWithPermLocation.getId()));
+  }
+
+  @Test
+  public void responseContainsAllRequiredHeaders() throws Exception {
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId, annexLibraryLocationId);
+
+    CompletableFuture<HttpClientResponse> createCompleted = new CompletableFuture<>();
+    Item item = buildItem(holdingsRecordId, null, null);
+
+    client
+      .post(InterfaceUrls.itemsStorageUrl(""), item,
+        StorageTestSuite.TENANT_ID, createCompleted::complete);
+
+    HttpClientResponse response = createCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(response.statusCode(), is(201));
+    assertThat(response.getHeader("location"), not(isEmptyString()));
+  }
+
+  private Item buildItem(UUID holdingsRecordId,
+                         UUID tempLocation, UUID permLocation) {
```

Comment:
buildItem uses "temp, perm" as the parameter order whereas createInstanceAndHolding uses "perm, temp".
This is inconsistent and makes it difficult to understand the unit tests.
I suggest changing buildItem to also use "perm, temp".

Reply:
fixed, thanks.